### PR TITLE
Update license to Apache 2

### PR DESCRIPTION
**Pull request by neerajprad**
Addresses https://github.com/pyro-ppl/pyro/issues/2249.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# Copyright (c) 2017-2020 Contributors to the Pyro project.
```
Comment (fritzo) [inline docs/source/conf.py]: Let's just use "Copyright Contributors to the Pyro project" as recommended by Steve Winslow

Comment (neerajprad) [reply docs/source/conf.py]: Okay, done.

File: numpyro/compat/util.py [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (fritzo) [inline numpyro/compat/util.py]: need an extra newline here for flake8

File: numpyro/patch.py [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (fritzo) [inline numpyro/patch.py]: ditto: add an extra newline

Comment (neerajprad) [reply numpyro/patch.py]: Oh I missed this, thanks for catching this. Will update.

File: setup.py [hunk 4]
```diff
@@ -1,3 +1,6 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
```
Comment (fritzo) [inline setup.py]: Can you also update setup.py as
```diff
- 'License :: OSI Approved :: MIT License',
+ 'License :: OSI Approved :: Apache License 2.0',
```

**Pull request merged**